scipy/signal/tests/test_signaltools.py

```diff
@@ -1919,6 +1918,29 @@ def test_numpy_fastpath(self):
         assert_allclose(correlate(a, b, mode='full'), [6, 17, 32, 23, 12])
         assert_allclose(correlate(a, b, mode='valid'), [32])
 
+    @pytest.mark.parametrize("mode", ["valid", "same", "full"])
+    @pytest.mark.parametrize("behind", [True, False])
+    def test_return_lags(self, mode, behind, return_lags=True):
+        # generate random data
+        x = np.random.random(1000)
```

Comment:
In general it's not a good idea to use un-seeded `rand` in tests. It's probably in principle fine here, but still probably better to do something like:
```
rng = np.random.RandomState(0)
x = rng.random(100)
...
y = np.concatenate([np.random.rand(100), x])
```

Reply:
Makes complete sense, thanks!